### ISSUE TITLE
[API Request] DiscordSendMessage / DiscordReceiveMessage events

### ISSUE THREAD

**Issue by Gunging** [DiscordSRV/DiscordSRV#919]
So one can make a plugin that parses a character sequence, lets say `:thinking:`, into an unused unicode character `㜁` when a player sends a chat. Then make a font texture pack, and *magic*, players can now use 🤔 in the mc chat.

**DiscordSRV** will then send the following:
![image](https://user-images.githubusercontent.com/48371009/100668516-a7f94280-3321-11eb-9b26-97d5d9ca9f5b.png)

Thus, currently, one must use a very hacky method of running an Async Task that restores the original message (that has `:thinking:` instead of `㜁`) before DiscordSRV runs its own Async Task that actually sends the message to the server. The result is the expected:
![image](https://user-images.githubusercontent.com/48371009/100668675-eabb1a80-3321-11eb-8e39-4bc1c3960b70.png)

However, its a bit uncomfortable to resort to this method, plus there is no way to intercept an incoming message to parse into `㜁`

So basically:

- _DiscordSendMessageEvent_ that runs right before DiscordSRV sends a message to discord (and one can get/set the message as you would the to the _AsyncPlayerChatEvent_ itself).
- _DiscordReceiveMessage_ that runs right before DiscordSRV broadcasts a message sent from discord (with same get/set functionality as described above).

**Comment by Vankka** [comment]
There are already events for this,

Discord -> Minecraft: `DiscordGuildMessagePostProcessEvent`
Minecraft -> Discord: `GameChatMessagePostProcessEvent`
